src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateMarcAuthorityEventHandler.java

```diff
@@ -73,29 +78,36 @@ public CreateMarcAuthorityEventHandler(Storage storage, MappingMetadataCache map
       var context = constructContext(dataImportEventPayload.getTenant(), dataImportEventPayload.getToken(), dataImportEventPayload.getOkapiUrl());
       prepareEvent(dataImportEventPayload);
       String jobExecutionId = dataImportEventPayload.getJobExecutionId();
-      String recordId = dataImportEventPayload.getContext().get(RECORD_ID_HEADER);
-      String chunkId = dataImportEventPayload.getContext().get(CHUNK_ID_HEADER);
-      mappingMetadataCache.get(jobExecutionId, context)
-        .map(parametersOptional -> parametersOptional.orElseThrow(() ->
-          new EventProcessingException(format(MAPPING_METADATA_NOT_FOUND_MSG, jobExecutionId,
-            recordId, chunkId))))
-        .onSuccess(mappingMetadata -> defaultMapRecordToAuthority(dataImportEventPayload, mappingMetadata))
-        .compose(v -> {
-          var authorityCollection = storage.getAuthorityRecordCollection(context);
-          final var authorityAsJson = prepareAuthority(dataImportEventPayload);
-          var authority = Json.decodeValue(authorityAsJson.encodePrettily(), Authority.class);
-          return addAuthority(authority, authorityCollection);
-        })
-        .onSuccess(createdAuthority -> {
-          LOGGER.info("Created an Authority record by jobExecutionId: '{}' and recordId: '{}' and chunkId: '{}' ", jobExecutionId,
-            recordId, chunkId);
-          payloadContext.put(AUTHORITY.value(), Json.encodePrettily(createdAuthority));
-          future.complete(dataImportEventPayload);
-        })
-        .onFailure(e -> {
-          LOGGER.error("Failed to save new Authority by jobExecutionId: '{}' and recordId: '{}' and chunkId: '{}' ",jobExecutionId,
-            recordId, chunkId, e);
-          future.completeExceptionally(e);
+      String recordId = payloadContext.get(RECORD_ID_HEADER);
+      String chunkId = payloadContext.get(CHUNK_ID_HEADER);
+
+      Future<RecordToEntity> recordToAuthorityFuture = idStorageService.store(recordId, UUID.randomUUID().toString(), dataImportEventPayload.getTenant());
```

Comment:
As I remember authority UUID already generated on the earlier stage, so we need to extract it from payload and use it as parameter for IdStorageService instead of generating new UUID

Reply:
That's right. This should be fixed.